### DISCUSSION TITLE
Router Component NotFound Fragment

### DISCUSSION THREAD

**Discussion by GuyCre8ive** [dotnet/aspnetcore#52120] [Q&A] [answered]
I wrote a web developer that would write pure HTML when a route was not found and I'm having a hard time figuring out how to redo this since NotFound was removed for some reason, any suggestions for where I can pass a page URL to do this now? I figured out how to redo this with an actual page URL but I'm still struggling to come up with a way to render the HTML for the home page.

**Comment by GuyCre8ive** [comment]
Well I thought I had a solution but none of my razor pages work now. This is frustrating..

**Comment by GuyCre8ive** [comment]
Wait, is Blazor SEO friendly now? Maybe I can use it for everything then, that's awesome!

**Reply by seangwright** [reply, marked as answer]
@GuyCre8ive Yes, you can [render an entire Blazor app server-side](https://chrissainty.com/blazor-in-dotnet-8-server-side-and-streaming-rendering/) using normal HTTP request/response (no websockets) and use Blazor as a replacement for most scenarios you would use MVC or Razor Pages for in the past.
